### DISCUSSION TITLE
Running image recognition/trail cam on same Rpi?

### DISCUSSION THREAD

**Discussion by tomwhite101** [mcguirepr89/BirdNET-Pi#517] [Ideas]
Hi All,
do people think that it would be possible to run [this](https://www.electromaker.io/project/view/whats-destroying-my-yard-pest-detection-with-raspberry-pi) type of set up on the same pi which is running Birdnet Pi - or would it be too much load/conflict?

I it would be very cool!

Cheers,
Tom

**Comment by mcguirepr89** [comment]
Howdy, Tom!!! This is a cool project indeed!!! I think it would be worth trying something like this out. Were I to try something like this, I would forego the PIR sensor and the Microsoft image processing, and instead opt to use the [Motion Project](https://motion-project.github.io/) to compare pixels instead of IR motion (just one less piece of hardware).

Motion (if not configured with hardware constraints in mind) can be super bloated, but can also be paired down to its (new) API and can be compiled with OpenCV (check out [Motionplus](https://github.com/Motion-Project/motionplus)) to add some AI image processing.

Resources will be the bottleneck, but certainly this is worth exploring.

**Reply by blogdemoi** [reply]
Pikrellcam https://billw2.github.io/pikrellcam/pikrellcam.html is a good project too, though not being developped anymore. I use it to record birds, video + audio. It can also record sounds only.

**Reply by nilspupils** [reply]
Great idea to combine camera and microphone detection!
I used:
https://github.com/interactionresearchstudio/NaturewatchCameraServer
for some time, they use motion detection (best way to go) and it is fun to use, but it is not actively developed and has some serious shortcomings. They use circular video recording, so they actually capture some seconds before the motion triggers, definitely a nice touch! They also do timelapse recording and i have captured some nices videos with it. 
Would be great to have it in BirdNET-Pi. 
Best, 
Nils

**Reply by matthew73210** [reply]
This maybe be on the limit of what an rpi4 can do, deffo a no for an rpi3. opi an easy yes. Nice idea though, as mentioned could even use the microphone audio source.
